Gaussian splatting support for Aria

brentyi opened the pull request:
Added support for the Fisheye624 model in the full image dataloader. @kerrj can you check if my changes make sense?

`ns-process-data aria` now also saves the Aria MPS points, which seems to improve results slightly.

Without points:
![aria_without_points](https://github.com/nerfstudio-project/nerfstudio/assets/6992947/3a8ea042-2303-4548-8db4-e6fd50bc71a2)

With points:
![aria_with_points](https://github.com/nerfstudio-project/nerfstudio/assets/6992947/123513c8-6c42-4e3b-b219-0136606d2283)

(this is the same capture as in https://github.com/nerfstudio-project/nerfstudio/pull/2617)


FYI @sweeneychris

sweeneychris commented on nerfstudio/data/datamanagers/full_images_datamanager.py:
```diff
@@ -414,3 +317,137 @@ def next_eval_image(self, step: int) -> Tuple[Cameras, Dict]:
+        map1 = unproj[..., 1]
+        map2 = unproj[..., 0]
+
+        image = cv2.remap(image, map1, map2, interpolation=cv2.INTER_LINEAR)
```
nit: might be better to use grid_interpolate?

brentyi replied:
- `grid_interpolate`: you mean switching to a PyTorch-based remapping? I think doing that would probably be better but we can leave for a future PR; the `cv2.remap()` implementation seems more consistent with the other camera models and we could refactor all of them together
- for using `projectaria_tools`: this seems doable but since we don't have access to the VRS files in the preprocessed dataset I think avoiding the dependency here makes sense!

sweeneychris commented on nerfstudio/scripts/datasets/process_project_aria.py:
```diff
@@ -219,6 +221,21 @@ def main(self) -> None:
 
+        # save global point cloud, which is useful for Gaussian Splatting.
+        points_path = self.mps_data_dir / "global_points.csv.gz"
+        if points_path.exists():
```
You should always filter the points from the aria point cloud --> here is a simple way to do it: https://facebookresearch.github.io/projectaria_tools/docs/data_utilities/core_code_snippets/mps#point-cloud

SeaOtocinclus replied:
We also have LOAD interface for any MPS data, no need to include pandas

i.e
```
points_data = mps.read_global_point_cloud(points_path) if points_path else None
# Filter out low confidence points
points_data = filter_points_from_confidence(points_data)
# Down sample points (optional)
# points_data = filter_points_from_count(points_data, 500_000)
# Retrieve point position
point_positions = [it.position_world for it in points_data]
```

brentyi replied:
done!!

filtered points (red) look reasonable:
![image](https://github.com/nerfstudio-project/nerfstudio/assets/6992947/7d6b5ff8-2208-4c5f-ba29-1a9d637283f1)

sweeneychris reviewed (commented):
overall LGTM and glad it was not too much surgery to get it working! Only real Q is about the choice for undistortion parameters (hardcoded vs programmatic). Note that not all Aria profiles have the same resolution for the RGB camera

SeaOtocinclus commented on nerfstudio/data/datamanagers/full_images_datamanager.py:
```diff
@@ -414,3 +317,137 @@ def next_eval_image(self, step: int) -> Tuple[Cameras, Dict]:
+        import warnings
+
+        warnings.warn("Fisheye624 support in the full images datamanager currently assumes data from Project Aria.")
+        undist_h = 500
+        undist_w = 500
+        undist_K = torch.eye(3)
+        undist_K[0, 0] = 150.0  # fx
+        undist_K[1, 1] = 150.0  # fy
```
Should not come from the camera calibration itself so it would work for various Aria recording data profile

kerrj replied:
agree, isn't it weird for these to be hardcoded?

brentyi commented:
Thanks for the feedback everyone! I added logic for computing the undistorted camera parameters, and fixed some issues with the masks.

Some more results:
![image](https://github.com/nerfstudio-project/nerfstudio/assets/6992947/a6b2efe2-5bab-4ef5-998b-57b17d0f3d5c)

The results look cleaner now:
![image](https://github.com/nerfstudio-project/nerfstudio/assets/6992947/491f9fd1-1d40-4f78-8f4a-6b0776da925b)

With initialization points (filtered from MPS) overlayed:
![image](https://github.com/nerfstudio-project/nerfstudio/assets/6992947/137f6c25-497a-4ee7-a9bb-02d46737dcb3)

sweeneychris commented on nerfstudio/data/datamanagers/full_images_datamanager.py:
```diff
@@ -416,3 +319,156 @@ def next_eval_image(self, step: int) -> Tuple[Cameras, Dict]:
+        undist_K = torch.eye(3)
+        undist_K[0, 0] = undistort_focal  # fx
+        undist_K[1, 1] = undistort_focal  # fy
+        undist_K[0, 2] = (undist_w - 1) / 2.0  # cx; for a 1x1 image, center should be at (0, 0).
```
I think it would make more sense to set the principle point of the rectified pinhole image with the principle point from the fisheye calibration. In other words, I'm not sure the principle point should move during rectification

brentyi replied:
Do you mean to replace `(undist_w - 1) / 2.0` directly with `camera.cx` (or equivalently, `fisheye624_params[3]`)?

To me that seems less compatible with how we're currently computing the undistorted image dimensions. The principal point would need to change if you decreased `fisheye_crop_radius` such that the the undistorted image was much smaller than the original distorted one, right?

f-dy commented:
Thank you for fixing the masks in GS! https://github.com/nerfstudio-project/nerfstudio/pull/2785/commits/d1783dfb5baf880395fae11f25a8f73517bc09df LGTM!

kerrj reviewed (approved):
lgtm!
